pkg/genericapiserver/config.go

```diff
@@ -423,7 +448,7 @@ func (c completedConfig) New() (*GenericAPIServer, error) {
 }
 
 // MaybeGenerateServingCerts generates serving certificates if requested and needed.
-func (c completedConfig) MaybeGenerateServingCerts(alternateIPs ...net.IP) error {
+func (c *Config) MaybeGenerateServingCerts(alternateIPs ...net.IP) error {
```

Comment:
Final solution, no, but it eliminates another level of defaulting.  I think we factor out one more and we can finally do this in order.